internal/test/net.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import "net"
+
+// borrowed from https://github.com/phayes/freeport
```

Comment:
This project is BSD-3 licensed. Please review the license terms.

https://github.com/phayes/freeport/blob/master/LICENSE.md

Because of the terms set out there, we cannot copy this code verbatim and claim copyright.

Reply:
Nice catch matt, missed that.

Reply:
@bacongobbler I copied this code from another test, so I'd imagine I should be able to use the same workaround as has been used there. I'll try to find that piece of code and use that instead, good catch!

Reply:
For reference: it's in the current codebase https://github.com/helm/helm/pull/7613/files#diff-80ccd988dcde8b4c1b8553b6061327fdL257-L271 I'll try pulling it in as a dependency